magniv/build/build.py

```diff
@@ -28,29 +28,43 @@ def _get_owner(root):
     return "local"
 
 
-def get_tasks_from_file(filepath: str, root: str, req: str, used_keys: Dict) -> Tuple[List, Dict]:
+def get_decorated_nodes(parsed_ast: ast.AST) -> List:
     """
-    > This function takes a filepath, root, req, and used_keys as arguments and returns a dictionary of
-    tasks and used_keys
+    It returns a list of all function and async function definitions in the AST that have at least one
+    decorator
 
-    :param filepath: The path to the file that contains the task
-    :param root: The root directory of the project
-    :param req: The location of the requirements.txt file
-    :param used_keys: This is a dictionary of keys that have been used so far. This is used to ensure
-    that no two tasks have the same key
-    :return: A list of dictionaries, each dictionary is a task.
+    Args:
+      parsed_ast (ast.AST): The parsed AST of the file.
+
+    Returns:
+      A list of nodes that are either a FunctionDef or AsyncFunctionDef and have a decorator_list.
     """
-    tasks = []
-    with open(filepath) as file:
-        parsed_ast = ast.parse(file.read())
-    decorated_nodes = [
+    return [
         node
         for node in ast.walk(parsed_ast)
         if (
             isinstance(node, (ast.FunctionDef, ast.AsyncFunctionDef))
             and len(node.decorator_list) > 0
         )
     ]
+
+
+def get_tasks_from_file(
```

Comment:
The name of this function is no longer accurate, right? this function is responsible for getting the magniv tasks from the ast.